test/framework/src/main/java/org/elasticsearch/test/ESTestCase.java

```diff
@@ -218,6 +224,7 @@ public abstract class ESTestCase extends LuceneTestCase {
     private static final Collection<String> loggedLeaks = new ArrayList<>();
 
     private HeaderWarningAppender headerWarningAppender;
+    private List<ThreadPoolExecutor> threadPoolExecutors = new ArrayList<>();
```

Comment:
I am thinking we could do something like we do now in Lucene, see https://github.com/apache/lucene/pull/12588 . We could create a single executor eagerly, and then reuse across all the searchers that are meant to get an executor.

Reply:
Thanks for the suggestion @javanna. I tried this and got some esoteric errors about thread leaks in some tests.  After a bunch of time hunting down the root cause, I realized that multiple executors can be created in a single test, and we need to keep track of all of them for cleanup purposes.

See for example `org.elasticsearch.xpack.core.security.authz.accesscontrol.SecurityIndexReaderWrapperIntegrationTests#testDLS` where ContextIndexSearcher is called in a `for` loop.